### PR TITLE
Handle either _text or _stext as the kernel relocation symbol.

### PR DESCRIPTION
**Pull request by s-kanev**
The perf tool can use either for the main kernel mapping and stores that
in the mapping name ('[kernel.kallsyms]_text' vs '[kernel.kallsyms]_stext').

With #675 the name of the relocation symbol is already available,
this is just the policy change to start using it.

Adds a unit test with a fake vmlinux ELF image.

### CODE REVIEW

File: internal/binutils/binutils.go [hunk 1]
```diff
@@ -440,8 +440,8 @@ func (b *binrep) openELF(name string, start, limit, offset uint64, relocationSym
 	}
 
 	var (
-		stextOffset *uint64
-		pageAligned = func(addr uint64) bool { return addr%4096 == 0 }
+		kernelOffset *uint64
+		pageAligned  = func(addr uint64) bool { return addr%4096 == 0 }
```
[the remark this answers is not in the document]
Comment (s-kanev) [reply internal/binutils/binutils.go:444]: Yes, gofmt tried to keep the block aligned after the variable above got one char longer.

File: internal/binutils/binutils.go [hunk 2]
```diff
@@ -455,10 +455,18 @@ func (b *binrep) openELF(name string, start, limit, offset uint64, relocationSym
 		if err != nil && err != elf.ErrNoSymbols {
 			return nil, err
 		}
+
+		// The kernel relocation symbol (the mapping start address) can be either
+		// _text or _stext. When profiles are generated by `perf`, which one was used is
+		// distinguised by the mapping name for the kernel image:
```
[the remark this answers is not in the document]
Comment (s-kanev) [reply internal/binutils/binutils.go:461]: Done, thanks.

File: internal/binutils/binutils_test.go [hunk 3]
```diff
@@ -838,3 +839,139 @@ func TestELFObjAddr(t *testing.T) {
...
+
+	symNames := buf{data: make([]byte, 0)}
+	syms := []elf.Sym64{
+		{}, // first symbol empty by convention
+		{Name: symNames.write("_text"), Info: 0, Other: 0, Shndx: 0, Value: textAddr, Size: 0},
+		{Name: symNames.write("_stext"), Info: 0, Other: 0, Shndx: 0, Value: stextAddr, Size: 0},
+	}
+
+	const numSections = 5
+	const textSize = 16 // # zero bytes we'll write for the contents of .text.
```
[the remark this answers is not in the document]
Comment (s-kanev) [reply internal/binutils/binutils_test.go:893]: Done.

**Review by gmarin13 — APPROVED** [review]
Thank you.

**Pull request merged**